### PR TITLE
Add ReplaceStringProcessor for simple string substitution that doesn't involve regex

### PR DESCRIPTION
**Pull request by dinujoh**
Add ReplaceStringProcessor for simple string substitution that doesn't involve regex

### Description
Add ReplaceStringProcessor for simple string substitution that doesn't involve regex. It uses java String.replace() method that is highly optimized for simple string replacements
 
### Issues Resolved
Resolves #[[4953](https://github.com/opensearch-project/data-prepper/issues/4953)]

### Check List
- [x] New functionality includes testing.
- [x] New functionality has a documentation issue. Please link to it in this PR.
  - [x] New functionality has javadoc added
- [x] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by dlvenable — DISMISSED** [review]
This is a nice improvement for pipeline authors to perform new mutations.

Thank you for using `EventKey`.

File: data-prepper-plugins/mutate-string-processors/src/main/java/org/opensearch/dataprepper/plugins/processor/mutatestring/ReplaceStringProcessorConfig.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+public class ReplaceStringProcessorConfig implements StringProcessorConfig<ReplaceStringProcessorConfig.Entry> {
+    public static class Entry {
+        @JsonPropertyDescription("The key to modify.")
+        private EventKey source;
+        @JsonPropertyDescription("The substring to be replaced in the source.")
+        private String from;
+        @JsonPropertyDescription("The string to be substituted for each match of `from`.")
+        private String to;
+
+        @JsonProperty("substitute_when")
```
Comment (oeyh) [inline data-prepper-plugins/mutate-string-processors/src/main/java/org/opensearch/dataprepper/plugins/processor/mutatestring/ReplaceStringProcessorConfig.java:28]: Should we call it `replace_when` here?

**Review by oeyh — COMMENTED** [review]
The changes look good! 

I wonder if we should merge this into the existing `substitute_string` processor as a different method. Something like this:
```yaml
    - substitute_string:
        entries:
          - source: "message"
            from: "ab"
            to: "ef"
            method: "simple"  # default is "regex" for existing replace using regex matcher
```

File: data-prepper-plugins/mutate-string-processors/src/main/java/org/opensearch/dataprepper/plugins/processor/mutatestring/ReplaceStringProcessorConfig.java (hunk 1 above)
Comment (dinujoh) [reply data-prepper-plugins/mutate-string-processors/src/main/java/org/opensearch/dataprepper/plugins/processor/mutatestring/ReplaceStringProcessorConfig.java]: good catch

**Comment by dlvenable** [comment]
> The changes look good!
> 
> I wonder if we should merge this into the existing `substitute_string` processor as a different method. Something like this:
> 
> ```yaml
>     - substitute_string:
>         entries:
>           - source: "message"
>             from: "ab"
>             to: "ef"
>             method: "simple"  # default is "regex" for existing replace using regex matcher
> ```

So if I follow you, maybe we don't need a new processor here? Just an option?

**Comment by dinujoh** [comment]
> The changes look good!
> 
> I wonder if we should merge this into the existing `substitute_string` processor as a different method. Something like this:
> 
> ```yaml
>     - substitute_string:
>         entries:
>           - source: "message"
>             from: "ab"
>             to: "ef"
>             method: "simple"  # default is "regex" for existing replace using regex matcher
> ```

Keeping them decoupled will help with simplicity.

**Comment by oeyh** [comment]
> So if I follow you, maybe we don't need a new processor here? Just an option?

Yes. That's the idea. It reduces duplicate code and is more extensible if we ever add other string replace methods in the future. But I'm OK with the new processor added here as well.

**Review by oeyh — APPROVED** [review]

**Review by dlvenable — APPROVED** [review]

**Pull request merged**